### PR TITLE
Begin debugger implementation - inproc BP support

### PR DESCRIPTION
**Pull request by plnelson**
Beginning of debugger support.  Add inproc breakpoint support.
Also needed to add a Dispose method to INes so circular dependencies don't prevent cleanup.

### CODE REVIEW

File: src/debug.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+#include "stdafx.h"
+#include "debug.h"
+
+int g_dbgEvent = 0;
+int g_dbgParam = 0;
+int g_dbgEnableLoadEvent = 0;
+
+void __declspec(noinline) DebuggerNotify(int dbgEvent, int param)
```
Comment (chuckries) [inline src/debug.cpp]: we'll need to make the __declspec `#define`'d to windows somehow.

**Comment by chuckries** [comment]
Is the plan for the debug service to stay in proc? Or would things like this be done in the DAC/concord component?

**Comment by chuckries** [comment]
Also how do you want to handle this cross platform?

**Comment by plnelson** [comment]
Most of the inproc and OPEC code will be the same.  The initial inproc implementation was good for testing.
There is no plan currently for cross platform.  We might be able to do Managed OPEC on *nix platforms, but at this point I don't know how feasible that will be.  I was planning to only support OPEC in Windows/x86 for now.

**Comment by chuckries** [comment]
I assumed there wouldn't be cross plat support. How do you want to isolate the in proc code so that we can still build cross plat?

**Pull request merged**